catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplication.java

```diff
@@ -112,17 +137,54 @@ public void init() {
                 .filter(Objects::nonNull)
                 .map(transformer::toFieldFilter)
                 .filter(Objects::nonNull)
+                .sorted(Comparator.comparing(CommonTemplate::getTitle))
                 .collect(Collectors.toList()),
         MAPPER::toJson);
 
+    put(
+        "/forms/query",
+        APPLICATION_JSON,
+        (req, res) ->
+            runWhenNotGuest(
+                res,
+                () ->
+                    doCreateOrUpdate(
+                        res,
+                        Stream.of(safeGetBody(req))
+                            .map(MAPPER::fromJson)
+                            .map(Map.class::cast)
+                            .map(transformer::toQueryTemplateMetacard)
+                            .filter(Objects::nonNull)
+                            .findFirst()
+                            .orElse(null))),
+        MAPPER::toJson);
+
+    put(
+        "/forms/result",
+        APPLICATION_JSON,
+        (req, res) ->
+            runWhenNotGuest(
+                res,
+                () ->
+                    doCreateOrUpdate(
+                        res,
+                        Stream.of(safeGetBody(req))
+                            .map(MAPPER::fromJson)
+                            .map(Map.class::cast)
+                            .map(transformer::toAttributeGroupMetacard)
+                            .filter(Objects::nonNull)
+                            .findFirst()
+                            .orElse(null))),
+        MAPPER::toJson);
+
     delete(
         "/forms/:id",
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
 
-          Subject subject = SecurityUtils.getSubject();
-          String currentUser = SubjectUtils.getName(subject);
+          Subject subject = (Subject) SecurityUtils.getSubject();
```

Comment:
Why is this casted?

Reply:
Because we want DDF's subject, not Shiro's subject.